engine/src/main/java/dgca/verifier/app/engine/DefaultCertLogicEngine.kt

```diff
@@ -73,27 +75,21 @@ class DefaultCertLogicEngine(private val jsonLogicValidator: JsonLogicValidator)
                     hcertVersion == null || ruleVersion == null || hcertVersion.first != ruleVersion.first -> Result.OPEN
                     hcertVersion.isGreaterOrEqualThan(ruleVersion) &&
                             jsonLogicValidator.isDataValid(
-                        rule.logic,
-                        dataJsonNode
-                    ) -> Result.PASSED
+                                rule.logic,
+                                dataJsonNode
+                            ) -> Result.PASSED
```

Comment:
An early return  could remove a full level of nesting and make this unwieldy indentation a bit easier to read.
```kotlin
override fun validate(
    certificateType: CertificateType,
    hcertVersionString: String,
    rules: List<Rule>,
    externalParameter: ExternalParameter,
    payload: String
): List<ValidationResult> {
    if (rules.isEmpty()) return emptyList()

    val validationResults = mutableListOf<ValidationResult>()
    val dataJsonNode = prepareData(externalParameter, payload)
    val hcertVersion = hcertVersionString.toVersion()
    rules.forEach { rule ->
        val ruleVersion = rule.version.toVersion()
        val res = when {
            hcertVersion == null || ruleVersion == null || hcertVersion.first != ruleVersion.first -> Result.OPEN
            hcertVersion.isGreaterOrEqualThan(ruleVersion) &&
                    jsonLogicValidator.isDataValid(
                        rule.logic,
                        dataJsonNode
                    ) -> Result.PASSED
            else -> Result.FAIL
        }
        val cur: String = affectedFieldsDataRetriever.getAffectedFieldsData(
            rule,
            dataJsonNode,
            certificateType
        )
        validationResults.add(
            ValidationResult(
                rule,
                res,
                cur,
                null
            )
        )
    }
    return validationResults
}
```